rs/ethereum/cketh/minter/src/deposit.rs

```diff
@@ -403,3 +169,186 @@ pub async fn update_last_observed_block_number() -> Option<BlockNumber> {
+                    INFO,
+                    "[scrape_contract_logs]: Failed to scrape {} logs in range {block_range}: {e:?}",
+                    S::display_id()
+                );
+                return;
+            }
+        }
+    }
+}
+
+async fn scrape_block_range<S>(
+    rpc_client: &EthRpcClient,
+    contract_address: Address,
+    topics: Vec<Topic>,
+    block_range: BlockRangeInclusive,
+) -> Result<(), MultiCallError<Vec<LogEntry>>>
+where
+    S: LogScraping,
+{
+    let mut subranges = VecDeque::new();
+    subranges.push_back(block_range);
+
+    while !subranges.is_empty() {
+        let range = subranges.pop_front().unwrap();
+        let (from_block, to_block) = range.clone().into_inner();
+
+        let request = GetLogsParam {
+            from_block: BlockSpec::from(from_block),
+            to_block: BlockSpec::from(to_block),
+            address: vec![contract_address],
+            topics: topics.clone(),
+        };
+
+        let result = rpc_client
+            .eth_get_logs(request)
+            .await
+            .map(<S::Parser>::parse_all_logs);
+
+        match result {
+            Ok((events, errors)) => {
+                register_deposit_events(S::display_id(), events, errors);
+                mutate_state(|s| S::update_last_scraped_block_number(s, to_block));
+            }
+            Err(e) => {
+                log!(
+                    INFO,
+                    "Failed to get {} logs in range {range}: {e:?}",
+                    S::display_id()
+                );
+                if e.has_http_outcall_error_matching(HttpOutcallError::is_response_too_large) {
+                    if from_block == to_block {
+                        mutate_state(|s| {
+                            process_event(
+                                s,
+                                EventType::SkippedBlockForContract {
+                                    contract_address,
+                                    block_number: to_block,
+                                },
+                            );
+                        });
+                        mutate_state(|s| S::update_last_scraped_block_number(s, to_block));
+                    } else {
+                        let (left_half, right_half) = range.partition_into_halves();
+                        if let Some(r) = right_half {
+                            let upper_range = subranges
+                                .pop_front()
+                                .map(|current_next| r.clone().join_with(current_next))
+                                .unwrap_or(r);
+                            subranges.push_front(upper_range);
+                        }
+                        if let Some(lower_range) = left_half {
+                            subranges.push_front(lower_range);
+                        }
+                        log!(
+                            INFO,
+                            "Too many logs received. Will with ranges {subranges:?}"
```

Comment:
"Will with ranges"?

Reply:
oops, I inadvertently deleted the keyword `retry`.